userspace/ksud/src/module.rs

```diff
@@ -346,21 +345,28 @@ pub fn install_module(zip: String) -> Result<()> {
     if !modules_img_exist && !modules_update_img_exist {
         // if no modules and modules_update, it is brand new installation, we should create a new img
         // create a tmp module img and mount it to modules_update
-        let result = Exec::shell(format!(
-            "dd if=/dev/zero of={} bs=1M count={}",
-            tmp_module_img, grow_size_per_m
-        ))
-        .stdout(subprocess::NullFile)
-        .stderr(subprocess::Redirection::Merge)
-        .join()?;
-        ensure!(result.success(), "create ext4 image failed!");
+        let result = File::create(tmp_module_img);
+        ensure!(
+            result.is_ok(),
+            "create ext4 image file failed: {}",
+            result.unwrap_err()
+        );
+        let result = result.unwrap().set_len(grow_size);
+        ensure!(
+            result.is_ok(),
+            "extend ext4 image failed: {}",
+            result.unwrap_err()
+        );
 
         // format the img to ext4 filesystem
-        let result = Exec::shell(format!("mkfs.ext4 {}", tmp_module_img))
-            .stdout(subprocess::NullFile)
-            .stderr(subprocess::Redirection::Merge)
-            .join()?;
-        ensure!(result.success(), "format ext4 image failed!");
+        let result = Command::new(format!("mkfs.ext4 {tmp_module_img}"))
+            .stdout(Stdio::null())
```

Comment:
we should also dummy the `stderr`

Reply:
`stderr` defaults to `piped` with `output`. When it is set to `null`, we will not able to get the output.

Reply:
I think that dummying stderr makes diagnosing problems harder.